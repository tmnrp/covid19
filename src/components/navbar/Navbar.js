import "./Navbar.scss";
import React from "react";
import { useHistory } from "react-router-dom";
import { PAGES_LIST } from "../../util/Constants";

let history;
const Navbar = () => {
  history = useHistory();
  return (
    <ul id="navbar" className="primary-border bg-panel-bg-color fade-in-down">
      {getPages()}
    </ul>
  );
};

const getPages = () => {
  return PAGES_LIST.map(page => {
    return (
      <li
        key={page}
        className="navbar-items-container bg-panel-bg-darken-hover"
        onClick={e => onNavItemClick(e, page)}
      >
        <div className="page-icon"></div>
        <div className="page-name-link primary-color">{page.toUpperCase()}</div>
      </li>
    );
  });
};

const onNavItemClick = (e, page) => {
  history.push(`/covid19/${page}`);
};

export default Navbar;
