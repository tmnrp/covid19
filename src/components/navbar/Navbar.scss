@import "../../index.scss";

#navbar {
  display: none;
  list-style: none;
}

#navbar.is-active {
  display: initial;
  position: absolute;
  right: 10px;
  top: 55px;
}

.navbar-items-container {
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
  cursor: pointer;
}

.page-name-link {
  text-decoration: none;
  padding: 10px 20px;
}
