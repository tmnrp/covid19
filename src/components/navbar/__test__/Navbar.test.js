import { Navbar, getPages } from "../Navbar";
import "@testing-library/react";
import "@testing-library/jest-dom";
import { fireEvent, render } from "@testing-library/react";

let isClicked = false;
const clickCallback = () => {
  isClicked = true;
};
const pages = [
  { name: "page 1", icon: "icon 1" },
  { name: "page 2", icon: "icon 2" }
];

test("Get pages renders properly", () => {
  expect(getPages(pages, clickCallback)).toMatchSnapshot();
});

test("Get pages click testing", () => {
  const { getByTestId } = render(
    Navbar({ pages: pages, clickCallback: clickCallback })
  );
  expect(isClicked).toBeFalsy();
  fireEvent.click(getByTestId("navbar").querySelector("li"));
  expect(isClicked).toBeTruthy();
});

test("Navbar renders properly", () => {
  expect(
    Navbar({ pages: pages, clickCallback: clickCallback })
  ).toMatchSnapshot();
});
