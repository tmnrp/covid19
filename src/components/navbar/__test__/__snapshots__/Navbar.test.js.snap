// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Get pages renders properly 1`] = `
Array [
  <li
    className="navbar-items-container bg-panel-bg-darken-hover"
    onClick={[Function]}
  >
    <div
      className="page-icon primary-color"
    >
      icon 1
    </div>
    <div
      className="page-name-link primary-color"
    >
      PAGE 1
    </div>
  </li>,
  <li
    className="navbar-items-container bg-panel-bg-darken-hover"
    onClick={[Function]}
  >
    <div
      className="page-icon primary-color"
    >
      icon 2
    </div>
    <div
      className="page-name-link primary-color"
    >
      PAGE 2
    </div>
  </li>,
]
`;

exports[`Navbar renders properly 1`] = `
<ul
  className="primary-border bg-panel-bg-color fade-in-down"
  data-testid="navbar"
  id="navbar"
>
  <li
    className="navbar-items-container bg-panel-bg-darken-hover"
    onClick={[Function]}
  >
    <div
      className="page-icon primary-color"
    >
      icon 1
    </div>
    <div
      className="page-name-link primary-color"
    >
      PAGE 1
    </div>
  </li>
  <li
    className="navbar-items-container bg-panel-bg-darken-hover"
    onClick={[Function]}
  >
    <div
      className="page-icon primary-color"
    >
      icon 2
    </div>
    <div
      className="page-name-link primary-color"
    >
      PAGE 2
    </div>
  </li>
</ul>
`;
